app/services/expression_evaluators/measure_unit.rb

```diff
@@ -71,5 +78,15 @@ def uk_measure_amounts
     def xi_measure_amounts
       @xi_measure_amounts ||= filtered_commodity(source: 'xi').applicable_measure_units
     end
+
+    def measure_condition
+      return nil if component.is_a? Api::MeasureComponent
+
+      measure.measure_conditions.detect do |measure_condition|
+        measure_condition.measure_condition_components.any? do |measure_condition_component|
+          measure_condition_component.as_json == component.as_json
```

Comment:
Casting to a json string seems an odd way of determining if its the measure condition component your looking for? Could this be a redefining of `==` on MeasureConditionComponent? and a more defined comparison on that?

Reply:
As json is fine but I could move the location of the comparison with an override of the == method in the measure_condition_component model